tests/devicelibs_test/disk_test.py

```diff
@@ -0,0 +1,91 @@
+def raise_other_error(*args):
+    raise OtherError("other message")
+
+
+class DiskLibTestCase(unittest.TestCase):
+    def test_lsm_dependency_guard(self):
+        """Validate handling of missing lsm dependency."""
+        # If lsm cannot be imported update_volume_info should yield an empty volume list.
+        with patch("blivet.devicelibs.disk._lsm_required._check_avail", return_value=False):
+            disklib.update_volume_info()
+            self.assertEqual(disklib.volumes, dict())
+
+    def test_lsm_error_handling(self):
+        """Validate handling of potential lsm errors."""
+        with patch("blivet.devicelibs.disk._lsm_required._check_avail", return_value=True):
+            with patch("blivet.devicelibs.disk.lsm") as _lsm:
+                _lsm.LsmError = FakeLsmError
+
+                # verify that we end up with an empty dict if lsm.Client() raises LsmError
+                _lsm.Client.side_effect = raise_lsm_error
+                disklib.update_volume_info()
+                self.assertEqual(disklib.volumes, dict())
+
+                # verify that any error other than LsmError gets raised
+                _lsm.Client.side_effect = raise_other_error
+                with self.assertRaises(OtherError):
+                    disklib.update_volume_info()
+
+    def test_update_volume_list(self):
+        """Validate conversion of lsm data."""
+        def client_capabilities(client, system):
+            if "jbod" in system.lower():
+                client.supported.return_value = False
+            else:
+                client.supported.return_value = True
+
+        lsm_volumes = [("Smart Array P840 in Slot 1",
+                        sentinel.RAID_TYPE_RAID0,
+                        262144,
+                        4,
+                        262144,
+                        1048576),
+                       ("LSI MegaRAID SAS",
+                        sentinel.RAID_TYPE_OTHER,
+                        524288,
+                        2,
+                        524288),
+                       ("Big Rig JBOD",
+                        None,
+                        None,
+                        None,
+                        None,
+                        None)]
+
+        with patch("blivet.devicelibs.disk._lsm_required._check_avail", return_value=True):
+            with patch("blivet.devicelibs.disk.lsm") as _lsm:
+                _lsm.Volume.RAID_TYPE_RAID0 = sentinel.RAID_TYPE_RAID0
+                _lsm.Volume.RAID_TYPE_OTHER = sentinel.RAID_TYPE_OTHER
+                _lsm.Capabilities.VOLUME_RAID_INFO = sentinel.VOLUME_RAID_INFO
+                _lsm.Client.capabilities.side_effect = client_capabilities
+                _lsm.Client.volumes.return_value = lsm_volumes
+                disklib.update_volume_info()
+                for lvol, bvol in zip(lsm_volumes, disklib.volumes):
+                    self.assertEqual(bvol.system, _lvol[0])
+                    if client_capabilities(Mock(), bvol.system):
```

Comment:
I believe you are missing import for `Mock`, e.g.  `from unittest.mock import patch, sentinel, Mock, Size`